sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1515,13 +1554,27 @@ def task_worker(self):
                             self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_FAILED
                             continue
                         host_lanes_mask = self.port_dict[lport]['host_lanes_mask']
-                        self.log_notice("{}: Setting lanemask=0x{:x}".format(lport, host_lanes_mask))
+                        self.log_notice("{}: Setting host_lanemask=0x{:x}".format(lport, host_lanes_mask))
+			
+                        self.port_dict[lport]['media_lane_count'] = int(api.get_media_lane_count(appl))
+                        self.port_dict[lport]['media_lane_assignment_options'] = int(api.get_media_lane_assignment_option(appl))
+                        media_lane_count = self.port_dict[lport]['media_lane_count']
+                        media_lane_assignment_options = self.port_dict[lport]['media_lane_assignment_options']
+                        self.port_dict[lport]['media_lanes_mask'] = self.get_cmis_media_lanes_mask(api,
+                                                                        appl, lport, subport)
+                        if self.port_dict[lport]['media_lanes_mask'] <= 0:
+                            self.log_error("{}: Invalid media lane mask received - media_lane_count {} subport {} "
```

Comment:
Why do we need to display media_lane_count here (wondering since we are not passing it as a parameter to get_cmis_media_lanes_mask)?

Reply:
As part of the get_cmis_media_lanes( ) api, we print these information as part of failure. 
But when the api is successful but return invalid data, we need to print the same information in the calling function.